.github/workflows/tests.yml

```diff
@@ -0,0 +1,289 @@
+name: Tests
+
+on:
+  push:
+    branches: [ 'develop', 'release-*' ]
+  pull_request:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toxenv:
+          - 'check-sampleconfig'
+          - 'check_codestyle'
+          - 'check_isort'
+          - 'mypy'
+          - 'packaging'
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+    - run: pip install tox
+    - run: tox -e ${{ matrix.toxenv }}
+
+
+  lint-crlf:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check line endings
+      run: scripts-dev/check_line_terminators.sh
+
+
+  lint-newsfile:
+    if: ${{ github.base_ref == 'develop' }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+    - run: pip install tox
+    - name: Patch Buildkite-specific test script
+      run: |
+        sed -i -e 's/\$BUILDKITE_PULL_REQUEST/${{ github.event.number }}/' \
+          scripts-dev/check-newsfragment
+    - run: scripts-dev/check-newsfragment
+
+
+  lint-sdist:
+    if: ${{ contains(github.ref, 'release-') }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
```

Comment:
What python version does this resolve to? 3.6?

I recommend setting this to the minimally-supported version, and/or do a matrix between the minimal + latest supported version.

Reply:
This resolves to the latest 3.x version; don't see a reason to matrix just for checking that wheels and sdists build... either they do or they don't.